src/python/gem5/resources/downloader.py

```diff
@@ -79,38 +80,20 @@ def _download(url: str, download_to: str, max_attempts: int = 6) -> None:
 
     # TODO: This whole setup will only work for single files we can get via
     # wget. We also need to support git clones going forward.
-
     attempt = 0
     while True:
         # The loop will be broken on a successful download, via a `return`, or
         # if an exception is raised. An exception will be raised if the maximum
         # number of download attempts has been reached or if a HTTP status code
         # other than 408, 429, or 5xx is received.
         try:
-            # check to see if user requests a proxy connection
             use_proxy = os.getenv("GEM5_USE_PROXY")
             if use_proxy:
-                # If the "use_proxy" variable is specified we setup a socks5
-                # connection.
-
-                import socket
-                import ssl
-
-                import socks
-
-                IP_ADDR, host_port = use_proxy.split(":")
-                PORT = int(host_port)
-                socks.set_default_proxy(socks.SOCKS5, IP_ADDR, PORT)
-                socket.socket = socks.socksocket
-
-                # base SSL context for https connection
-                ctx = ssl.create_default_context()
-                ctx.check_hostname = False
-                ctx.verify_mode = ssl.CERT_NONE
-
                 # get the file as a bytes blob
                 request = urllib.request.Request(url)
-                with urllib.request.urlopen(request, context=ctx) as fr:
+                with urllib.request.urlopen(
+                    request, context=get_proxy_context()
+                ) as fr:
```

Comment:
You could be a little neater here. Checking the environment variable isn't necessary:

```python
# check to see if user requests a proxy connection.
# `get_proxy_context() returns the proxy context if the user requested use of one, else `None`.
proxy_context = get_proxy_context()
if proxy_context:
    # get the file as a bytes blob
    request = urllib.request.Request(url)
    with urllib.request.urlopen(request, context=proxy_context()) as fr:
    # ...
```

Reply:
Or, `with urllib.request.urlopen(request, context=proxy_context) as fr:` rather.